library/MachineControl.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+from urllib.parse import quote
+
+from library.common import *
+from library.NetworkControl import NetworkControl
+from library.StateControl import StateControl
+
+class MachineControl(object):
+    def __init__(self, requester, machine_name):
+        self.requester = requester
+
+        try:
+            self.machine_id = self._get_id(machine_name)
+            self.networkcontrol = NetworkControl(self.requester, self.machine_id)
+        except Exception as err:
+            raise err
+
+        self.statecontrol = StateControl(self.requester, self.machine_id)
+
+    def _get_id(self, machine_name):
+        """ Look up machine by name """
+        q = '(= name "{}")'.format(quote(machine_name))
+        url = "/api/v1/machine?q={}&show_all=false".format(q)
+        try:
+            data = self.requester.get(url)
+        except Exception as err:
+            raise err
+
+        return data[0]['id']
+
+    def get_network_info(self, command, interface_name):
```

Comment:
This doesn't make sense in machine. Why not use the NetworkControl directly where needed?

Reply:
I thought it'd be useful keeping a 'Machine' object abstraction in the client, which could keep track of its state and which would centralize all operations related to machines.

Reply:
But you're adding network logic to the machine class's usage of the network object that the network class doesn't have. That's an anti-pattern. :)

Reply:
MachineControl now acts only as a proxy, not routing logic at all.